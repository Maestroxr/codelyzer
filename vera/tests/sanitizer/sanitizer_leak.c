#include <stdlib.h>
#include <stdio.h>
int main(int argc, char** argv) 
{
	int res = -1;
	/*
		Leak Sanitizer
	*/
	void* p = (void*)malloc(7);
	if (!p)
	{
		//do something one
	}
	p = 0; // The memory is leaked here.	-V-
	p = (void*)malloc(7);
	if (!p)
	{
		//do something one
	}
	p = 0; // The memory is leaked here.	-V-
	/***/
	
	return res;
}