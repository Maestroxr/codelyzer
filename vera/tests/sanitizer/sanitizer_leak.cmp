C:\Work\AutoAssessment\codelyzer\vera\tests\sanitizer\sanitizer_leak.c:1: Memory:[C:\Work\AutoAssessment\codelyzer\vera\tests\sanitizer\sanitizer_leak]: SUMMARY: LeakSanitizer: 14 byte(s) leaked in 2 allocation(s).
C:\Work\AutoAssessment\codelyzer\vera\tests\sanitizer\sanitizer_leak.c:9: Memory:[C:\Work\AutoAssessment\codelyzer\vera\tests\sanitizer\sanitizer_leak]: Detected memory leaks, suspect variables: [p]
		#1 sanitizer_leak.c:15 in function 'main'
C:\Work\AutoAssessment\codelyzer\vera\tests\sanitizer\sanitizer_leak.c:15: Memory:[C:\Work\AutoAssessment\codelyzer\vera\tests\sanitizer\sanitizer_leak]: Detected memory leaks, suspect variables: [p]
		#1 sanitizer_leak.c:15 in function 'main'
