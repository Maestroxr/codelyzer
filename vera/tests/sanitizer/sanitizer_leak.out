C:\Work\AutoAssessment\codelyzer\vera\tests\sanitizer\sanitizer_leak.c:9: Memory:[C:\Work\AutoAssessment\codelyzer\vera\tests\sanitizer\sanitizer_leak]: detected memory leaks, Direct leak, suspect variable identifiers: p.
C:\Work\AutoAssessment\codelyzer\vera\tests\sanitizer\sanitizer_leak.c:15: Memory:[C:\Work\AutoAssessment\codelyzer\vera\tests\sanitizer\sanitizer_leak]: detected memory leaks, Direct leak, suspect variable identifiers: p.
